internal/server/middlewares.go

```diff
@@ -143,17 +154,6 @@ func requestLogger(lg *zap.Logger) middleware {
 
 			next.ServeHTTP(fr, req)
 
-			fields := []zap.Field{
-				zap.String("request_path", req.URL.Path),
-				zap.String("request_method", req.Method),
-				zap.String("request_id", req.Header.Get(headerRequestID)),
-				zap.String("client_id", clientID),
-				zap.String("trace_id", span.SpanContext().TraceID.String()),
-				zap.String("response_time", time.Since(t).String()),
-				zap.String("request_body", string(jsonBody)),
-				zap.Int("status", wrapped.Status),
-			}
-
 			if !is2xx(wrapped.Status) {
```

Comment:
Rather than using wrappedwriter, can we just define a func to log and defer call the function?

Reply:
I am not sure why it was implemented this way. @spy16 any idea?
Also, it does not seem to be in the scope of this PR, CMIIW? Let's take this in a separate refactoring story?

Reply:
Oh okay I thought this part of this changes. But still we should not think about moving this to a “refactoring” story as much as we could. That is just introduce a tech debt which should be able to be avoided asap.

Reply:
Yeah makes sense.
@StewartJingga @sudheerpal ^